agent/consul/catalog_endpoint.go

```diff
@@ -204,6 +204,10 @@ func servicePreApply(service *structs.NodeService, authz ACLResolveResult, authz
 		}
 	}
 
+	if service.Connect.SidecarService != nil && service.Connect.SidecarService.Port == 0 {
```

Comment:
This will generally never be hit because SidecarService will always be nil.

https://github.com/hashicorp/consul/blob/a668c369306a1441377bbdaa1970d61fd5121b6f/agent/agent_endpoint_test.go#L3649-L3652

So I think the better validation would be to check the top level Port (or I guess SocketPath) to ensure one of those is filled in for services with kind connect-proxy.

Reply:
Removed this check.